plugins/batterystat.koplugin/main.lua

```diff
@@ -0,0 +1,205 @@
+
+local DataStorage = require("datastorage")
+local KeyValuePage = require("ui/widget/keyvaluepage")
+local PowerD = require("device"):getPowerDevice()
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local T = require("ffi/util").template
+local logger = require("logger")
+local _ = require("gettext")
+
+local State = {}
+
+function State:new(o)
+    o = o or {}
+    setmetatable(o, self)
+    self.__index = self
+    o:validate()
+    return o
+end
+
+function State:validate()
+    if self.percentage == nil or self.timestamp == nil then
+        self.percentage = PowerD:getCapacity()
+        self.timestamp = os.time()
+    end
+end
+
+local Usage = {}
+
+function Usage:new(o)
+    o = o or {}
+    setmetatable(o, self)
+    self.__index = self
+    o:validate()
+    return o
+end
+
+function Usage:validate()
+    if self.percentage == nil or self.time == nil then
+        self.percentage = 0
+        self.time = 0
+    end
+end
+
+function Usage:append(state)
+    local state2 = State:new()
+    self.percentage = self.percentage + (state2.percentage - state.percentage)
+    self.time = self.time + os.difftime(state2.timestamp - state.timestamp)
+end
+
+function Usage:minutes()
+    return self.time / 60
+end
+
+function Usage:hours()
+    return self:minutes() / 60
+end
+
+function Usage:percentagePerHour()
+    if self.time == 0 then
+        return math.huge
+    else
+        return self.percentage / self:hours()
+    end
+end
+
+local function shorten(number)
+    return string.format("%.2f", number);
+end
+
+function Usage:dump(kv_pairs)
+    table.insert(kv_pairs, {_("    Consumed %"), shorten(self.percentage)})
```

Comment:
It looks a little bit better, as "%" is not so significant. Well, it's my personal option.